senders/new_sender.go

```diff
@@ -28,20 +28,20 @@ func NewSender(wfURL string, setters ...Option) (Sender, error) {
 	} else {
 		sender.internalRegistry = sdkmetrics.NewNoOpRegistry()
 	}
-	sender.pointHandler = newLineHandler(metricsReporter, cfg, internal.MetricFormat, "points", sender.internalRegistry)
-	sender.histoHandler = newLineHandler(metricsReporter, cfg, internal.HistogramFormat, "histograms", sender.internalRegistry)
-	sender.spanHandler = newLineHandler(tracesReporter, cfg, internal.TraceFormat, "spans", sender.internalRegistry)
-	sender.spanLogHandler = newLineHandler(tracesReporter, cfg, internal.SpanLogsFormat, "span_logs", sender.internalRegistry)
-	sender.eventHandler = newLineHandler(metricsReporter, cfg, internal.EventFormat, "events", sender.internalRegistry)
 
+	hf := internal.NewHandlerFactory(
```

Comment:
Does it make sense to also include cfg.BatchSize here?

Reply:
I moved that one around a couple of times... I put it on the factory constructor at first, but then when I realized that Events have a special case for batch size, I liked the idea of expressing that through the API by letting the other handlers accept a batch size, but not that one.